TemplateMethod/src/test/java/devslp/designpatterns/TemplateMethod/SortStrategyExample/SortStrategyTest.java

```diff
@@ -0,0 +1,21 @@
+package devslp.designpatterns.TemplateMethod.SortStrategyExample;
+
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class SortStrategyTest {
+    @Test
+    public void testBubbleSorter()
+    {
+        Object [] theArray = {5, 1, 4, 2, 8};
+        DoubleSortHandler intSortHandler = new DoubleSortHandler();
+        BubbleSorter bubbleSorter = new BubbleSorter(intSortHandler);
+        bubbleSorter.sort(theArray);
+        assertEquals(intSortHandler.getArray()[0], 1);
```

Comment:
Using DoubleSortHandler, the elements in the sorted array will be converted to double, so I suggest to change every integer constant  (second parameter) to a double constant, i.e. assertEquals(intSortHandler.getArray()[0], **1.0**);